server/src/main/java/org/elasticsearch/monitor/os/OsProbe.java

```diff
@@ -42,6 +42,23 @@
 import java.util.regex.Pattern;
 import java.util.stream.Collectors;
 
+/**
+ * The {@link OsProbe} class retrieves information about the physical and swap size of the machine
+ * memory, as well as the system load average and cpu load.
+ *
+ * In some exceptional cases, it's possible the underlying native method used by
+ * {@link #getFreePhysicalMemorySize()} and {@link #getTotalPhysicalMemorySize()} can return a
+ * negative value. Because of this, we prevent those methods from returning negative values,
+ * returning 0 instead.
+ *
+ * The OS can report a negative number in a number of cases:
+ * - Non-supported OSes (HP-UX, AIX, OSX after failing to initialize host statistics
```

Comment:
This is slightly misleading since we support macOS (the official name of the OS).

Reply:
I updated the wording to make it clearer, thanks